src/ape_ethereum/ecosystem.py

```diff
@@ -42,6 +46,23 @@
 }
 
 
+class ProxyType(IntEnum):
+    Minimal = 0  # eip-1167 minimal proxy contract
+    Standard = 1  # eip-1967 standard proxy storage slots
+    Beacon = 2  # eip-1967 beacon proxy
+    UUPS = 3  # # eip-1822 universal upgradeable proxy standard
+    Vyper = 4  # vyper <0.2.9 create_forwarder_to
+    Clones = 5  # 0xsplits clones
+    GnosisSafe = 6
+    OpenZeppelin = 7  # openzeppelin upgradeability proxy
+
+
+@dataclass
+class ProxyInfo:
```

Comment:
This might be better off living alongside `EcosystemAPI` as a standard type

Reply:
That would make the `ProxyType` enum kind of complicated though?
What if an ecosystem defines new types?

Reply:
Oh, yeah fair

I think instead of `dataclass` it should maybe use `pydantic.BaseModel` then so it can serialize